src/main/groovy/jenkins/automation/utils/CheckmarxSecurityJobBuilder.groovy

```diff
@@ -0,0 +1,63 @@
+package jenkins.automation.utils
+
+import javaposse.jobdsl.dsl.DslFactory
+import javaposse.jobdsl.dsl.Job
+
+
+class CheckmarxSecurityJobBuilder {
+
+    String name
+    String description
+    String baseUrl
+    String scanRepo
+    String checkmarxComment
+    Boolean vulnerabilityThresholdEnabled
+    String highThreshold
+    String mediumThreshold
+    String lowThreshold
+
+    Job build(DslFactory factory){
+        factory.job(name){
+            it.description this.description
+            
+            scm {
+                git {
+                    remote {
+                        name('Repo to Scan')
+                        url(scanRepo)
+                    }
+                    
+                    branch('*/master')
+                    wipeOutWorkspace(true)
```

Comment:
there should be a less brute-force way to do this. check out the git options... I think there's one for simply reverting any changes. And really, I can't think of why a checkmarx job would ever have changes made to its workspace, so I'm really dubious that this would be necessary or would add value